howso/upgrade.amlg

```diff
@@ -381,6 +396,75 @@
 		(call !Return)
 	)
 
+	#!UpgradeFromSubTrainee
```

Comment:
will need a nice comment block about how this flow works from a high level on this method since it's recursive

Reply:
Only sorta recursive now... recursive across entities? I'll still comment lol